src/test/kotlin/com/makarytskyi/rentcar/repository/OrderRepositoryTest.kt

```diff
@@ -27,141 +27,170 @@ internal class OrderRepositoryTest : ContainerBase {
     @Test
     fun `create should insert order and return it with id`() {
         // GIVEN
-        val car = carRepository.create(randomCar())
-        val user = userRepository.create(randomUser())
+        val car = carRepository.create(randomCar()).block()
+        val user = userRepository.create(randomUser()).block()
+        val order = randomOrder(car?.id, user?.id).copy(id = null)
 
         // WHEN
-        val order = orderRepository.create(randomOrder(car.id, user.id).copy(id = null))
+        val createdOrder = orderRepository.create(order)
 
         // THEN
-        val foundOrder = orderRepository.findById(order.id.toString())
-        assertNotNull(foundOrder)
-        assertNotNull(order.id)
+        StepVerifier.create(createdOrder)
+            .assertNext {
+                assertNotNull(it.id)
+                assertEquals(order.copy(id = it.id), it)
+            }
+            .verifyComplete()
     }
 
     @Test
     fun `findAll should find all orders`() {
         // GIVEN
-        val car1 = carRepository.create(randomCar())
-        val user1 = userRepository.create(randomUser())
-        val order1 = orderRepository.create(randomOrder(car1.id, user1.id))
+        val car1 = carRepository.create(randomCar()).block()
+        val user1 = userRepository.create(randomUser()).block()
+        val order1 = orderRepository.create(randomOrder(car1?.id, user1?.id)).block()
 
-        val car2 = carRepository.create(randomCar())
-        val user2 = userRepository.create(randomUser())
-        val order2 = orderRepository.create(randomOrder(car2.id, user2.id))
+        val car2 = carRepository.create(randomCar()).block()
+        val user2 = userRepository.create(randomUser()).block()
+        val order2 = orderRepository.create(randomOrder(car2?.id, user2?.id)).block()
 
         // WHEN
-        val orders = orderRepository.findAll(0, 20)
+        val allOrders = orderRepository.findFullAll(0, 20)
 
         // THEN
-        assertTrue(orders.any { it.id == order1.id })
-        assertTrue(orders.any { it.id == order2.id })
+        StepVerifier.create(allOrders.collectList())
+            .assertNext { orders ->
```

Comment:
you could change it to containsAll

Reply:
They have different types: MongoOrder and AggregatedMongoOrder. I can't compare them in that way, so I compare them by fields.